agent/app/agent.go

```diff
@@ -99,9 +99,10 @@ const (
 	asgLifecyclePollWait           = time.Minute
 	asgLifecyclePollMax            = 120 // given each poll cycle waits for about a minute, this gives 2-3 hours before timing out
 
-	// the size of the buffer for the metric/task health/instance health telemetry channels.
-	// This values needs to be tuned after all channel related implementation is done.
-	telemetryChannelDefaultBufferSize = 15
+	// By default, TCS (or TACS) will reject metrics that are older than 5 minutes. Since our metrics collection interval
+	// is currently set to 20 seconds, setting a buffer size of 15 for each task and targeting at 1000 tasks (currently known maximum ~500)
+	// allows us to store exactly 5 minutes of metrics in these buffers in the case where we temporarily lose connect to TCS.
```

Comment:
Yeah, this was the behavior we discussed.

Reply:
Okay I see. 

One more follow-up. In the case where we restore a tacs session, and there are buffered metrics on the channel, will we try to send them all at once? It looks like each metrics object on the channel will be a separate request. Do we have any concerns for the burst traffic?

Reply:
Hmm I think I made a mistake here, this value should be a smaller number (like 15 in Fargate agent side).

This value is actually the number of times we call `getInstanceMetrics`, which grabs task metrics from all tasks/containers on the instance. I was setting this value based on 1000 tasks but it should actually not be part of the consideration, as no matter how many tasks we have on an instance, we will have them all in one `ecstcs.TelemetryMessage` object and sent through the channel.

After changing this value to 15, however, the number of request we sent to TACS after a session restoration will not change too much, as we are packing the request to 10 tasks each. That means, in a worst case scenario that does not take TACS down into consideration, that would mean we send about 750 (with 500 tasks) request from one single agent - 500 tasks * 15 sample count / 10 tasks per request. TACS's throttle limit is 999 for this case (for each connection) and therefore it would not be an issue.

Send followup link offline